src/adapter/http/src/smart_protocol/axum_server_pull_protocol.rs

```diff
@@ -110,7 +141,7 @@ impl AxumServerPullProtocolInstance {
         let head = metadata_chain
             .resolve_ref(&BlockRef::Head)
             .await
-            .map_err(|e| PullServerError::Internal(e.int_err()))?;
+            .protocol_int_err(TransferPhase::Pull(PullPhase::InitialRequest))?;
```

Comment:
Nice!

To make this even better you can change the signature:
```rust
fn protocol_int_err(phase: Into<TransferPhase>)
```
so you could call it as `.protocol_int_err(PullPhase::InitialRequest)` without needing wrap into enum manually.